src/ApiService/ApiService/Functions/TimerWorkers.cs

```diff
@@ -62,13 +65,24 @@ public class TimerWorkers {
 
         var nodes = _nodeOps.SearchStates(states: NodeStateHelper.NeedsWorkStates);
         await foreach (var node in nodes) {
-            _log.Info($"update node: {node.MachineId:Tag:MachineId}");
-            _ = await _nodeOps.ProcessStateUpdate(node);
+            try {
+                _log.Info($"updating node: {node.MachineId:Tag:MachineId} - state: {node.State:Tag:NodeState}");
+                var newNode = await _nodeOps.ProcessStateUpdate(node);
+                _log.Info($"completed updating node - now in state {newNode.State:Tag:NodeState}");
+            } catch (Exception ex) {
+                _log.Exception(ex, $"failed to process node");
+            }
         }
 
         var scalesets = _scaleSetOps.SearchAll();
         await foreach (var scaleset in scalesets) {
-            await ProcessScalesets(scaleset);
+            try {
+                _log.Info($"updating scaleset: {scaleset.ScalesetId:Tag:ScalesetId} - state: {scaleset.State:Tag:ScalesetState}");
+                var newScaleset = await ProcessScalesets(scaleset);
+                _log.Info($"completed updating scaleset - now in state {newScaleset.State:Tag:ScalesetState}");
+            } catch (Exception ex) {
+                _log.Exception(ex, $"failed to process scaleset");
```

Comment:
log names ? but I guess it's ok to dismiss those too....

Reply:
We can get the sequence of logs by correlation ID, so I don't want to repeat too much? But I can add it in.